src/main/scala/io/opentargets/etl/backend/drug/DrugCommon.scala

```diff
@@ -223,22 +223,16 @@ object DrugCommon extends Serializable with LazyLogging {
 
   def getUniqTargetsAndDiseasesPerDrugId(evidenceDf: DataFrame): DataFrame =
     evidenceDf
-      .filter(col("sourceId") === "chembl")
       .groupBy(col("drugId"))
       .agg(
-        collect_set(col("targetId")).as("targets"),
-        collect_set(col("diseaseId")).as("diseases")
+        collect_set(col("targetFromSource")).as("targets"),
+        collect_set(col("diseaseFromSource")).as("diseases")
```

Comment:
Thanks for this :) Just a comment, by extracting these entities from these fields we would not have resolvable IDs; whereas currently, in the API, the content of `rows` are IDs inside the OT indexes.
For example, `targetFromSource` will be either ChEMBL IDs and `diseaseFromSource` will be a disease label.
- You can fix the disease by using `diseaseFromSourceMappedId` instead (however we might find terms here that are outside of the disease index). 
- This is however not possible for the target, as the most resolvable ID that we have is the Uniprot ID inside `targetFromSourceId`

Does this make sense?

Reply:
Yes. Thanks for the catch!